chalk-solve/src/clauses/builtin_traits.rs

```diff
@@ -30,3 +30,19 @@ pub fn add_builtin_program_clauses<I: Interner>(
         WellKnownTrait::DropTrait => (),
     }
 }
+
+/// Check the same trait for a list of types.
```

Comment:
This comment and function name doesn't mean much to me, I'm afraid. It'd be good to give an example of how/why it is used.

Reply:
I am still not very at ease with chalk terminology, that's why I came up with such a poor comment. @Areredify could you help me out providing a better description for the helper function? Thanks!

Reply:
Uhh, something like "pushes a clause of the form `Implemented(T: Trait) :- Implemented(T1: Trait), ... Implemented(Tn: Trait)`, i.e. to prove `trait_ref` you need to prove the same trait bound on `tys`."

Reply:
OK, maybe something like:

Given a trait ref `T0: Trait` and a list of types `U0..Un`, pushes a clause of the form

```
Implemented(T0: Trait) :- Implemented(U0: Trait) .. Implemented(Un: Trait)
```

Reply:
Done!